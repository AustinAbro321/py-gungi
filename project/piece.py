from dataclasses import dataclass
from enum import Enum
from color import Color


class PieceType(Enum):
    MAJOR_GENERAL = 1
    LIEUTENANT_GENERAL = 2
    GENERAL = 3
    ARCHER = 4
    KNIGHT = 5
    MUSKETEER = 6
    CAPTAIN = 7
    SAMURAI = 8
    FORTRESS = 9
    CANNON = 10
    SPY = 11
    PAWN = 12
    KING = 13


@dataclass
class Piece:
    piece_type: PieceType
    color: Color


class PieceStack:
    def __init__(self,pieces: list[Piece] = []):
        self.pieces = pieces

    def pop(self) -> Piece:
        return self.pieces.pop()

    # def __str__(self) -> str:
    #     name = ""
    #     for piece in self.pieces:
    #         name += str(piece.piece_type.value)
    #     return name

    def __repr__(self) -> str:
        name = "piece"
        for piece in self.pieces:
            name += str(piece.piece_type.value)
        return name