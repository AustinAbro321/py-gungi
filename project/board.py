from piece import PieceStack


def starting_board() -> list[list[PieceStack]]:
    return [
        [PieceStack(),PieceStack(),PieceStack(),PieceStack(),
        PieceStack(),PieceStack(),PieceStack(),PieceStack(),PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()],
        [PieceStack(), PieceStack(), PieceStack(), PieceStack(),
        PieceStack(), PieceStack(), PieceStack(), PieceStack(), PieceStack()]]


class Board:
    def __init__(self, content: list[list[PieceStack]]):
        self.content = content

    def __repr__(self):
        name = ""
        for line in self.content:
            name += "["
            for pieceStack in line:
                name += pieceStack.__repr__() + ","
            name += "]"
            name += "\n"
        return name
