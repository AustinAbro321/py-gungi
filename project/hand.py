from piece import Piece, PieceType
from color import Color


def starting_hand(player: Color):
    hand = []
    hand.extend([Piece(PieceType.MAJOR_GENERAL, player)] * 4)
    hand.extend([Piece(PieceType.LIEUTENANT_GENERAL, player)] * 4)
    return hand
    # Piece(PieceType.LIEUTENANT_GENERAL, player), Piece(PieceType.LIEUTENANT_GENERAL, player),
    # Piece(PieceType.LIEUTENANT_GENERAL, player), Piece(PieceType.LIEUTENANT_GENERAL, player),
    # Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player),
    # Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player), Piece(PieceType.GENERAL, player),
    # Piece(PieceType.Archer, player), Piece(PieceType.Archer, player),
    # Piece(PieceType.Knight, player), ]


class Hand:
    def __init__(self, player: Color):
        self.player = Color.WHITE
        self.hand = starting_hand(player)
        print(starting_hand(player))
