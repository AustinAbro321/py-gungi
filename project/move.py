import typing
from dataclasses import dataclass
import copy
from board import Board


@dataclass
class Square:
    row: int
    col: int

@dataclass
class MoveInfo:
    point_a: Square
    point_b: Square

def translate_move() -> MoveInfo:
    return MoveInfo(Square(0,0),Square(0,1))

def move(board : Board,move_info: MoveInfo) -> Board:
    new_board = board
    point_a = move_info.point_a
    point_b = move_info.point_b
    new_board.content[point_b.row][point_b.col] = copy.deepcopy(board.content[point_a.row][point_a.col])
    board.content[point_a.row][point_a.col].pop()
    new_board.content[point_a.row][point_a.col] = board.content[point_a.row][point_a.col]
    return new_board
