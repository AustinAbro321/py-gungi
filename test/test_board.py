from approvaltests import verify

from board import starting_board, Board
from color import Color
from hand import starting_hand
from move import MoveInfo, Square, move
from piece import Piece, PieceType, PieceStack


def test_board_init():
    assert True


def test_color():
    assert Color.WHITE == Color.WHITE


def test_starting_hand():
    assert starting_hand(Color.WHITE) == [Piece(PieceType.MAJOR_GENERAL, Color.WHITE),
                                          Piece(PieceType.MAJOR_GENERAL, Color.WHITE),
                                          Piece(PieceType.MAJOR_GENERAL, Color.WHITE),
                                          Piece(PieceType.MAJOR_GENERAL, Color.WHITE)]

def test_move_approvals():
    board = Board(starting_board())
    piece_stack = PieceStack([Piece(PieceType.MAJOR_GENERAL, Color.WHITE)])
    board.content[0][0] = piece_stack
    move_info = MoveInfo(Square(0,0),Square(0,1))
    verify(move(board,move_info))