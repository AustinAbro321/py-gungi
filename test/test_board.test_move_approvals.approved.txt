[piece,piece1,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
[piece,piece,piece,piece,piece,piece,piece,piece,piece,]
